File: be-service/app/routers/price_analysis.py
from fastapi import APIRouter

from app.models.price_analysis import (
    ProductBenchmarkRequest,
    ProductPriceRecommendationRequest
)

from app.controllers.price_analysis_controller import (
    analyze_existing_product,
    recommend_price
)

router = APIRouter(
    prefix="/price-analysis",
    tags=["Price Analysis"]
)

@router.post(
    "/existing-product"
)
def analyze_product(
    payload:
    ProductBenchmarkRequest
):

    return {
        "success": True,
        "message":
            "Price analysis generated successfully",
        "data":
            analyze_existing_product(
                payload
            )
    }

@router.post(
    "/new-product"
)
def recommend_new_price(
    payload:
    ProductPriceRecommendationRequest
):

    return {
        "success": True,
        "message":
            "Price recommendation generated successfully",
        "data":
            recommend_price(
                payload
            )
    }

File: fe-service/src/lib/priceAnalysis.ts
import api from "./api";

export async function analyzeExistingProduct(
  productId: number
) {
  const response = await api.post(
    "/api/v1/price-analysis/existing-product",
    {
      product_id: productId,
    }
  );

  return response.data.data;
}

export async function recommendNewProductPrice(
  productName: string,
  basePrice: number
) {
  const response = await api.post(
    "/api/v1/price-analysis/new-product",
    {
      product_name: productName,
      base_price: basePrice
    }
  );

  return response.data.data;
}